enhancements/machine-config/alignment-with-image-mode-rhel-for-the-mco.md

```diff
@@ -0,0 +1,205 @@
+
+### API Extensions
+
+[TBD] We don’t anticipate any user facing API extension changes as of now. We will update this section if something comes up during the smoke tests, implementation or future planning.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+[TBD] We don’t anticipate any impact on Hypershift as of now. We will update this section if something comes up during the smoke tests, implementation or future planning.
+
+#### Standalone Clusters
+
+[TBD] We don’t anticipate any impact on Standalone Clusters as of now. We will update this section if something comes up during the smoke tests, implementation or future planning.
+
+#### Single-node Deployments or MicroShift
+
+[TBD] We don’t anticipate any impact on Single-node Deployments or MicroShift as of now. We will update this section if something comes up during the smoke tests, implementation or future planning.
+
+### Implementation Details/Notes/Constraints
+
+Phase 1: Please refer to On Cluster Layering enhancement for more details  
+Phase 2: [TBD] 
+Phase 3: 
+  * Setup bootc go-bindings for status reading and command execution 
+  * Replace rpm-ostree rebase by bootc switch for image rollout 
+
+### Risks and Mitigations
+
+[TBD]
+
+### Drawbacks
+
+[TBD]
+
+## Open Questions [optional]
+
+* Is Go binding for bootc a necessity for adapting a bootc update path in the MCO? 
+    - Q: Currently, bootc does not have go binding which will make it hard to read its status and deployment. It seems that we will need to create a Go library for client side interactions with bootc (similar to: https://github.com/coreos/rpmostree-client-go) Colin has suggested a way to auto-generate it (see comment in https://issues.redhat.com/browse/MCO-1026), which would be the most ideal solution. Workarounds should also be available, making go binding a non-goal item for this enhancement. 
+    - A: Go bindings for bootc is not a must and, as a result, will be outside of the scope of this enhancement and will not be implemented. The main purpose of creating go bindings for rpm-ostree/bootc is to have an easier way to parse the created json and to read the status of rpm-ostree/bootc. This can be done via creating a simple wrapper function inside of the MCO. It will make sense to separate it out to a standalone helper/library in the future when the demand raises, but is a non-goal for now.
+
+* Discussion question for gaps highlighted in phase 2 
+    - Q: A problem with the current model is that configs that involve file writing to local file system directories are not respected by “rpm-ostree rebase” / “bootc switch”. For example, if we write an SSH key to “/home/core/.ssh/authorized_keys'' directory during an OS image build and use rpm-ostree to rebase onto the newly built OS image, that file will not actually be there when the node reboots. Will this be handled/considered in the future planning of RHCOS and bootc? 
```

Comment:
This is related to my earlier comment about `/var`.

Reply:
Fixed based on the comment earlierhttps://github.com/containers/bootc/issues/76